src/transaction/TransactionController.ts

```diff
@@ -713,21 +732,49 @@ export class TransactionController extends BaseController<
       throw new Error('No sign method defined.');
     }
 
-    const gasPrice = getIncreasedPriceFromExisting(
+    // gasPrice (legacy non EIP1559)
+    const minGasPrice = getIncreasedPriceFromExisting(
       transactionMeta.transaction.gasPrice,
       CANCEL_RATE,
     );
 
+    const gasPirceFromValues =
+      isGasPriceValue(gasValues) && gasValues?.gasPrice;
```

Comment:
Nit: One branch can be removed here, as the condition before this guarantees that `gasValues` is truthy
```suggestion
      isGasPriceValue(gasValues) && gasValues.gasPrice;
```

Reply:
I found a bunch more similar instances of this - most of the `?` operators introduced are not necessary it seems.